### DISCUSSION TITLE
Non-connectable devices

### DISCUSSION THREAD

**Discussion by jakobwarnhjelm** [hbldh/bleak#1391] [Q&A] [answered]
I'm looking for an example how to get started when listening to data from non-connectable devices.
Like an thermometer or smart scale without pairing.

I managed to get the discovery example working using `BleakScanner` to get the UUID (MacOS).
However, I can't figure out how to connect using the `BleakClient` class.

Something like an infinte loop that prints the payload in hex would help a lot.

**Comment by dlech** [comment, marked as answer]
If the devices are non-connectable, I would expect that you cannot connect to them so BleakClient isn't going to be useful in this case. You can use the detection callback of BleakScanner to operate on the advertising data. There is an example in the examples folder here.

**Reply by jakobwarnhjelm** [reply]
Ok that seems reasonable!
